### PR TITLE
FMWK-252 Refactor query creator

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 1]
```diff
@@ -142,136 +134,203 @@ private Object convertIfNecessary(Object obj) {
...
+            qualifier = processOther(part, value1, value2, property, op, fieldName);
+        }
+
+        return new AerospikeCriteria(qualifier);
+    }
+
+    private Qualifier processId(Object value1) {
+        Qualifier qualifier;
+        if (value1 instanceof Collection<?>) {
+            qualifier = forIds(((Collection<?>) value1).toArray(String[]::new));
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:156]: Unsafe transition from a collection of Object to an array of String.

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 2]
```diff
@@ -142,136 +134,203 @@ private Object convertIfNecessary(Object obj) {
...
+        } else {
+            qualifier = forId((String) value1);
+        }
+        return qualifier;
+    }
+
+    private Qualifier processCollection(Part part, Object value1, Object value2, Iterator<?> parameters,
+                                        FilterOperation op, String fieldName) {
+        String dotPath = null;
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:165]: dotPath is always null. Consider passing an explicit null as an argument.

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 3]
```diff
@@ -289,46 +348,45 @@ private String getFieldName(String segmentName, AerospikePersistentProperty prop
...
             qualifiers = new Qualifier[params.size() / 2]; // keys/values qty must be even
             for (int i = 0, j = 0; i < params.size(); i += 2) {
                 setQbValuesForMapByKey(qb, params.get(i), params.get(i + 1));
-                qualifiers[j++] = setQualifierBuilderValues(qb, fieldName, op, part, params.get(i),
-                    null, null, dotPath).build();
+                qualifiers[j++] = setQualifier(qb, fieldName, op, part, params.get(i),
+                    null, null, dotPath);
             }
 
             return new AerospikeCriteria(Qualifier.and(qualifiers));
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:369]: Why wrapping with AerospikeCriteria?

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 4]
```diff
@@ -289,46 +348,45 @@ private String getFieldName(String segmentName, AerospikePersistentProperty prop
...
+                                                       Part part, String fieldName, FilterOperation op,
+                                                       String dotPath, boolean containingMapKeyValuePairs) {
         Qualifier[] qualifiers;
         if (containingMapKeyValuePairs) {
             qualifiers = new Qualifier[params.size() / 2]; // keys/values qty must be even
             for (int i = 0, j = 0; i < params.size(); i += 2) {
                 setQbValuesForMapByKey(qb, params.get(i), params.get(i + 1));
-                qualifiers[j++] = setQualifierBuilderValues(qb, fieldName, op, part, params.get(i),
-                    null, null, dotPath).build();
+                qualifiers[j++] = setQualifier(qb, fieldName, op, part, params.get(i),
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:365]: Increment j in the increment section of the for statement.

**Comment by roimenashe** [comment]
Nothing to add other than @reugn's comments

**Review by reugn — APPROVED** [review]

**Pull request merged**